src/main/java/dev/morling/onebrc/CalculateAverage_thomaswue.java

```diff
@@ -223,18 +194,47 @@ else if ((long) nameLength >= 4) {
                     }
                 }
             }
-
-            // Skip new line.
-            scanPtr++;
         }
     }
 
+    // Special method to convert a number in the specific format into an int value without branches created by
+    // Quan Anh Mai.
+    private static int convertIntoNumber(int decimalSepPos, long numberWord) {
+        int shift = 28 - decimalSepPos;
+        // signed is -1 if negative, 0 otherwise
+        long signed = (~numberWord << 59) >> 63;
+        long designMask = ~(signed & 0xFF);
+        // Align the number to a specific position and transform the ascii code
+        // to actual digit value in each byte
+        long digits = ((numberWord & designMask) << shift) & 0x0F000F0F00L;
+
+        // Now digits is in the form 0xUU00TTHH00 (UU: units digit, TT: tens digit, HH: hundreds digit)
+        // 0xUU00TTHH00 * (100 * 0x1000000 + 10 * 0x10000 + 1) =
+        // 0x000000UU00TTHH00 +
+        // 0x00UU00TTHH000000 * 10 +
+        // 0xUU00TTHH00000000 * 100
+        // Now TT * 100 has 2 trailing zeroes and HH * 100 + TT * 10 + UU < 0x400
+        // This results in our value lies in the bit 32 to 41 of this product
+        // That was close :)
+        long absValue = ((digits * 0x640a0001) >>> 32) & 0x3FF;
+        long value = (absValue ^ signed) - signed;
+        return (int) value;
+    }
+
+    private static int findDelimiter(long word) {
+        long input = word ^ 0x3B3B3B3B3B3B3B3BL;
+        long tmp = (input & 0x7F7F7F7F7F7F7F7FL) + 0x7F7F7F7F7F7F7F7FL;
+        tmp = ~(tmp | input | 0x7F7F7F7F7F7F7F7FL);
+        return Long.numberOfTrailingZeros(tmp) >>> 3;
+    }
```

Comment:
Can be micro-optimized as follows:
```java
private static int findDelimiter(long word) {
    long input = word ^ 0x3B3B3B3B3B3B3B3BL;
    long tmp = (input - 0x0101010101010101L) & ~input & 0x8080808080808080L;
    return Long.numberOfTrailingZeros(tmp) >>> 3;
}
```
Which saves 1 instruction.
Before:                  `139,978,244,139 instructions:u`
With this patch: `134,391,924,137 instructions:u`

Reply:
Since all characters before the delimiter are larger than `0x3B`, you can get away with

    long mask = (word - 0x3C3C3C3C3C3C3C3CL) & 0x8080808080808080L;
    return Long.numberOfTrailingZeros(mask) >>> 3;

Also you need to load this in a little endian manner.

Reply:
Yes, indeed, thanks. This got another almost 2%.

Reply:
@gunnarmorling This is already the second contribution from @mukel, I would like to ask if you could add him as a co-author to the submission as well in case you will re-evaluate.